app/models/shelter.rb

```diff
@@ -16,6 +16,12 @@ def self.order_by_number_of_pets
       .order("pets_count DESC")
   end
 
+  def self.shelters_apps_in_progress
+    pets = Pet.find(AppsPet.pluck(:pet_id))
+    shelter_ids = pets.pluck(:shelter_id)
+    shelters = Shelter.where(id: shelter_ids)
```

Comment:
Does this query actually verify that a shelter has an app that is "Pending"? Consider something like this: `Shelter.joins(pets: :applications).where("applications.status = 'Pending'")`